src/ImageSharp/Formats/Gif/GifEncoderCore.cs

```diff
@@ -144,22 +147,52 @@ private void EncodeGlobal<TPixel>(Image<TPixel> image, QuantizedFrame<TPixel> qu
                 }
                 else
                 {
-                    using (QuantizedFrame<TPixel> paletteQuantized = palleteQuantizer.CreateFrameQuantizer(() => quantized.Palette).QuantizeFrame(frame))
+                    QuantizedFrame<TPixel> paletteQuantized = palleteQuantizer.CreateFrameQuantizer(() => quantized.Palette).QuantizeFrame(frame);
+
+                    if (lastFrameDisposalMethod == DisposalMethod.NotDispose)
+                    {
+                        this.DoCutRepeatedPixelsGlobal(renderedFrame, paletteQuantized, transparencyIndex);
+                    }
+                    else
                     {
-                        this.WriteImageData(paletteQuantized, stream);
+                        renderedFrame.Dispose();
+                        renderedFrame = paletteQuantized;
```

Comment:
Pretty sure this is incorrect for `DisposalMethod.RestoreToBackground` and `RestoreToPrevious` since the quantized palette may have transparent pixels. Not entirely sure how to handle this, because I don't have a good grip on which pixels are used when `DisposalMethod`s are mixed.

Reply:
This is only an issue if the image frames are already optimized for GIF encoding by the user. So if they take advantage of the different disposal methods and transparency. Should that be supported or can I assume each frame stands on its own while encoding?

Reply:
Okay, thinking about this some more it really doesn't matter :p If pixels are already transparent we can't optimize them anyway. So I think this is completely fine. Still gotta write the tests though!